stats/system_tag.go

```diff
@@ -0,0 +1,120 @@
+package stats
+
+import (
+	"bytes"
+	"encoding/json"
+	"strings"
+)
+
+// SystemTagSet is a bitmask that is used to keep track
+// which system tags should be included with which metrics.
+type SystemTagSet uint32
+
+// SystemTagMap is a string to bool map (for lookup efficiency) that is used to keep track
+// which system tags should be included with with metrics.
+type SystemTagMap map[string]bool
```

Comment:
Shouldn't this be simply `TagMap` or, even better, a `TagSet`. Since this is meant to be used when there's a mix between system and non-system tags, it doesn't make sense to contain `System` in the name.

Reply:
Fair enough, done.